cli/medperf/entities/cube.py

```diff
@@ -115,30 +115,38 @@ def get(cls, cube_uid: str) -> "Cube":
         Returns:
             Cube : a Cube instance with the retrieved data.
         """
-        "Retrieve from local storage if cube already there"
         logging.debug(f"Retrieving the cube {cube_uid}")
         comms = config.comms
-        local_cube = list(
-            filter(lambda cube: str(cube.uid) == str(cube_uid), cls.all())
-        )
-        if len(local_cube) == 1:
-            logging.debug("Found cube locally")
-            return local_cube[0]
-
-        meta = comms.get_cube_metadata(cube_uid)
-        cube = cls(meta)
-        attempt = 0
-        while attempt < config.cube_get_max_attempts:
-            logging.info(f"Downloading MLCube. Attempt {attempt + 1}")
-            cube.download()
-            if cube.is_valid():
-                cube.write()
-                return cube
-            attempt += 1
-        logging.error("Max download attempts reached")
+
+        # Try to download the cube first
+        try:
+            meta = comms.get_cube_metadata(cube_uid)
+            cube = cls(meta)
```

Comment:
missing a `cube.download()` after this line. Currently, `cube.is_valid()` will always return False at the first attempt.

Reply:
This is by design. `cube.is_valid()` will fail if the required files are not present (which would be the case in a first-run). But, if the cube has already been downloaded before, we don't want to force to re-download images and additional files which are expensive. By checking validity first, we ensure already-downloaded skip the download procedure

Reply:
oh right that's perfect!